### ISSUE TITLE
--cluster-init  start cluster and https: raft_init(): io: io_submit: invalid argument 

### ISSUE THREAD

**Issue by damozhiying** [rancher/k3s#1283]
**Version:**
v1.17.0-rc.4-k3s.1
CentOS Linux release 7.5.1804 (Core) 
```
sudo ./k3s --debug server K3S_TOKEN=SECRET --docker --cluster-init
DEBU[0000] Asset dir /var/lib/rancher/k3s/data/8c8e1aca909a242c0d5e37efa7967ff7b6fe603880885b4f4e2374a7048b29ae 
DEBU[0000] Running /var/lib/rancher/k3s/data/8c8e1aca909a242c0d5e37efa7967ff7b6fe603880885b4f4e2374a7048b29ae/bin/k3s-server [./k3s --debug server K3S_TOKEN=SECRET --docker --cluster-init] 
INFO[2020-01-09T13:33:37.492646531+08:00] Starting k3s v1.17.0-rc.4-k3s.1 (25f50824)   
FATA[2020-01-09T13:33:37.523631733+08:00] starting kubernetes: preparing server: start cluster and https: raft_init(): io: io_submit: invalid argument 
```

**Comment by damozhiying** [comment]
**version**
v1.0.0 (18bd921c)
CentOS Linux release 7.5.1804 (Core)
```
sudo ./k3s1 --debug server K3S_TOKEN=SECRET --docker --cluster-init
INFO[0000] Preparing data dir /var/lib/rancher/k3s/data/2492a28b74abfbdbd3f943c8109a93e979669647bbbe95eadae43b60c667df69 
DEBU[0002] Running /var/lib/rancher/k3s/data/2492a28b74abfbdbd3f943c8109a93e979669647bbbe95eadae43b60c667df69/bin/k3s-server [./k3s1 --debug server K3S_TOKEN=SECRET --docker --cluster-init] 
INFO[2020-01-09T15:34:53.951011728+08:00] Starting k3s v1.0.0 (18bd921c)               
04:23:08.397 [DEBUG]: data dir: /var/lib/rancher/k3s/server/db/state.dqlite
04:23:08.397 [DEBUG]: metadata1: version 0, term 0, voted for 0
04:23:08.397 [DEBUG]: metadata2: version 0, term 0, voted for 0
04:23:08.397 [DEBUG]: metadata: version 2, term 0, voted for 0
04:23:08.397 [ERROR]: probe I/O capabilities: can't allocate write buffer
FATA[2020-01-09T15:34:54.264735146+08:00] starting kubernetes: failed to create task object 
```

**Comment by damozhiying** [comment]
**below is too simple,how to step by step?**
High Availability with Embedded DB (Experimental) (https://rancher.com/docs/k3s/latest/en/installation/ha-embedded/)

**Comment by davidnuzik** [comment]
Hello, while we expect K3s to work on CentOS 7.x, please take care to note that it is not officially supported. For a list of supported operating systems please reference our documentation here: https://rancher.com/docs/k3s/latest/en/installation/node-requirements/#operating-systems

**Comment by davidnuzik** [comment]
Regarding the question about the embedded (experimental) HA you can leverage the binary or the curl script. For example you could do the following with the curl script:

First server:
`curl -fL https://get.k3s.io | K3S_TOKEN='SECRET' sh -s - server --cluster-init`

Additional servers (3 minimum for HA):
`curl -fL https://get.k3s.io | K3S_TOKEN='SECRET' K3S_URL='https://MASTER_IP_OR_DNS:6443' sh -s - server `

possible may need to add to tls-san in some situations.

**Comment by davidnuzik** [comment]
FWIW Hussein has mentioned seeing this all the time on Ubuntu 16.04 LTS x86; moved to our next up milestone (v1.17.x) for consideration.

**Comment by erikwilson** [comment]
Related #1130

**Comment by galal-hussein** [comment]
https://github.com/canonical/raft/pull/113/files

**Comment by definitiontv** [comment]
I just  compiled from latest main branch and from chrisfleming fork
Still getting 

"starting kubernetes: preparing server: start cluster and https: raft_init(): io: io_submit: invalid argument"

On an original ubuntu 16.04 which was dist-upgraded beforehand to latest kernel.

Works fine on other instances and it is only the latest release that gives this error.

I have 8 or 9 similar  cloud instances - i  will try  all to see If I can  work out what works and what doesn't and post a list if there is any pattern.

**Comment by definitiontv** [comment]
..update - mea culpa - i didn't notice the dev branch.  i can confirm the package upgrade fixes this on Ubuntu 16.04.3 Thanks a lot!!
